pkg/injector/patch.go

```diff
@@ -138,41 +139,68 @@ func addContainer(target, add []corev1.Container, basePath string) (patch []JSON
 			path += "/-"
 		}
 		patch = append(patch, JSONPatchOperation{
-			Op:    "add",
+			Op:    addOperation,
 			Path:  path,
 			Value: value,
 		})
 	}
 	return patch
 }
 
-func updateAnnotation(target, add map[string]string, basePath string) (patch []JSONPatchOperation) {
-	for key, value := range add {
-		if target == nil {
-			// First one will be a Create
-			target = map[string]string{}
-			patch = append(patch, JSONPatchOperation{
-				Op:   "add",
-				Path: basePath,
-				Value: map[string]string{
-					key: value,
-				},
-			})
-		} else {
-			// Update
-			op := "add"
-			if target[key] != "" {
-				op = "replace"
-			}
-			patch = append(patch, JSONPatchOperation{
-				Op:    op,
-				Path:  path.Join(basePath, escapeJSONPointerValue(key)),
-				Value: value,
-			})
+func createTarget(key, path, value string) JSONPatchOperation {
+	return JSONPatchOperation{
+		Op:   addOperation,
+		Path: path,
+		Value: map[string]string{
+			key: value,
+		},
+	}
+}
+
+func patchTarget(key, basePath, patchOp, value string) JSONPatchOperation {
+	return JSONPatchOperation{
+		Op:    patchOp,
+		Path:  path.Join(basePath, escapeJSONPointerValue(key)),
+		Value: value,
+	}
+}
+
+// getSortedKeys returns the keys of a map in sorted string slice.
+// The purpose of this is to provide determinism when iterating over a map.
+func getSortedKeys(m map[string]string) []string {
+	keys := make([]string, 0, len(m))
+	for key := range m {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+	return keys
+}
+
+func updateAnnotation(target, add map[string]string, basePath string) []JSONPatchOperation {
+	var patches []JSONPatchOperation
+
+	// If the target does not exist we need to create it
+	noTarget := len(target) == 0
+
+	// We iterate over the sorted keys in order to have a deterministic output from this function.
```

Comment:
Why does the output from this function need to be deterministic? If it is for the purpose of testing, I would add this to the comment.